packages/react-compose/src/next/getSlots.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import { getNativeElementProps } from '@uifabric/utilities';
+import { GenericDictionary } from './types';
+import { nullRender } from './nullRender';
+
+/**
+ * Given the state and an array of slot names, will break out `slots` and `slotProps`
+ * collections. Ensures that each slot is a renderable component, and each slotProp is
+ * a list of acceptable native properties for the given slot. Native properties are only
+ * filtered if the slot is rendered "as" a string type, and will use `getNativeElementProps`
+ * to pass along only valid props. If the slot is a non-string value, no filtering is done.
+ *
+ * @param state - State including slot definitions
+ * @param slotNames - Name of which props are slots
+ * @returns An object containing the `slots` map and `slotProps` map.
+ */
+export const getSlots = (state: GenericDictionary, slotNames?: string[] | undefined) => {
+  const slots: GenericDictionary = {
+    root: state.as || nullRender,
+  };
+  const slotProps: GenericDictionary = {
+    root: getNativeElementProps(state.as, state),
+  };
+
+  if (slotNames) {
+    for (const name of slotNames!) {
```

Comment:
Do you have to do the `!`? Doesn't the condition above take care of that?

Reply:
fixed.